src/niri.rs

```diff
@@ -3934,6 +3935,10 @@ impl Niri {
             self.queue_redraw_all();
         }
     }
+
+    pub fn ipc_workspaces(&self) -> Arc<Mutex<Vec<IpcWorkspace>>> {
+        Arc::new(Mutex::new(self.layout.ipc_workspaces()))
```

Comment:
The other arc-mutexes in the IPC Context struct point at long-lived data that is constantly updated on any change. This function here on the other hand returns a newly constructed piece of data that is never updated, so it can simply return it as a `Vec<IpcWorkspace>`.

(Or actually, I don't think this function is needed because you can just call the layout method directly for now.)

Reply:
I agree... I originally didn't know what i was getting myself into when I started hacking away, so I made it "match" the signature of the other props. I should have re-worked that when I better understood what I was doing.